packages/array-range/index.d.ts

```diff
@@ -0,0 +1,3 @@
+declare function range(start: number, end: number|null, step?: number|null): number[];
```

Comment:
I need to improve the runtime type checking in my implementation because right now technically all 3 arguments are optional (`range()` is valid and returns an empty array) and all 3 arguments can also be `null` or `undefined` too.

In practice there is no point using null/undefined for `start` or `end` so I think something like this will suffice:
`declare function range(start?: number, end?: number|null|undefined, step?: number): number[];`

Reply:
I'm a bit unclear on what to do here. I agree that your suggested function signature makes sense, but it will prevent `range(null)` from passing the test, which you flagged below.. 

This would work to allow the `range(null)` case to pass. 

```typescript
declare function range(start?: number|null, end?: number|null|undefined, step?: number): number[];
```

But since, as you pointed out, there's no point in actually passing `null` here, maybe it makes more sense to just keep `range(null)` as an error and use your suggested signature:

```typescript
declare function range(start?: number, end?: number|null|undefined, step?: number): number[];
```

What do you think?

Reply:
Ah I see your point yeah. In that case I think the latter solution is better. Thanks for digging in to that!